providers/provider_default.go

```diff
@@ -33,14 +33,22 @@ var (
 )
 
 // GetLoginURL with typical oauth parameters
-func (p *ProviderData) GetLoginURL(redirectURI, state, _ string) string {
+// codeChallenge and codeChallengeMethod are the PKCE challenge and method to append to the URL params.
+//   they will be empty strings if no code challenge should be presented
+func (p *ProviderData) GetLoginURL(redirectURI, state, _, codeChallenge, codeChallengeMethod string) string {
 	extraParams := url.Values{}
+	if codeChallenge != "" && codeChallengeMethod != "" {
```

Comment:
You have this as optional yet if I understood the code in the main package correctly, it's always set right?

Reply:
This would only be set if the provider data `CodeChallengeMethod` is set to a hashing function. (plain or S256). If it is not set then we do not attempt PKCE and do not pass the extra URL argument